src/__changelog/mod.rs

```diff
@@ -0,0 +1,76 @@
+//! # Changelog and migrations
+//!
+//! All notable changes to this project will be documented in this module,
+//! along with migration instructions for larger changes.
+
+// Put the current changelog entry at the top of rustdoc output.
+pub use unreleased as current;
+
+#[doc = include_str!("unreleased.md")]
+pub mod unreleased {}
```

Comment:
Maybe we should mention in `CONTRIBUTING.md` specifically that we have to remove the `unreleased` module and re-export the current version as `current`.